atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -258,19 +267,16 @@ def resolve_signing_intent(self):
         self.compose_config.set_signing_intent(signing_intent['name'])
 
     def forward_composes(self):
-        repos_by_arch = {}
+        repos_by_arch = defaultdict(list)
         # set overrides by arch if arches are available
         for compose_info in self.composes_info:
-            for arch in compose_info.get('arches') or []:
-                repos_by_arch.setdefault(arch, [])
+            for arch in compose_info.get('arches', '').split(" "):
```

Comment:
Minor:
If 'arches' is not present, we'll end up running through this loop once with `''` for arch (because `''.split(' ')` is `['']`). If we want to handle 'arches' not being present, I'd suggest:
```
result_repofile = compose_info['result_repofile']
try:
    arches = compose_info['arches']
except KeyError:
    repos_by_arch[None].append(result_repofile)
else:
    for arch in arches.split(' '):
        repos_by_arch[arch].append(result_repofile)
```

But first: do we really need to handle 'arches' not being present in the compose response?

Reply:
ODCS API is a little odd when it comes to `sigkeys` and `arches` values. Instead of returning a list of strings, it returns a white-space separated string. I'd suggest using the same approach that was used for handling `sigkeys`:

```python
keys = compose_info.get('sigkeys', [])
....
if isinstance(keys, six.text_type):
    keys = keys.split()            
keys = set(keys)                   
```

This has the benefit of being future ready if ODCS ever decides to return the correct data type.

Note that it's better to not pass a parameter to `split`:
```python
>>> ''.split(' ')
['']
>>> ''.split()
[]
```

Reply:
Fixed for both.